File: auth-service/src/main/java/com/eirsteir/coffeewithme/authservice/domain/Role.java
package com.eirsteir.coffeewithme.authservice.domain;

import java.io.Serializable;
import javax.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@Data
@Accessors(chain = true)
@NoArgsConstructor
@Entity
public class Role implements Serializable {

  private static final long serialVersionUID = -3444524593791786238L;

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  private Long id;

  @Column(unique = true)
  private RoleType type;

  public Role(RoleType type) {
    this.type = type;
  }
}

File: auth-service/src/main/kotlin/com/eirsteir/coffeewithme/authservice/domain/Role.kt
package com.eirsteir.coffeewithme.authservice.domain

import java.io.Serializable
import javax.persistence.*


@Entity
data class Role(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private val id: Long = -1L,
    @Column(unique = true)
    val type: RoleType = RoleType.ROLE_USER,
) : Serializable {

    companion object {
        private const val serialVersionUID = -3444524593791786238L
    }
}